packages/editor/src/components/provider/index.js

```diff
@@ -243,6 +244,11 @@ export const ExperimentalEditorProvider = withRegistryProvider(
 			setCurrentTemplateId( template?.id );
 		}, [ template?.id, setCurrentTemplateId ] );
 
+		// Sets the right rendering mode when loading the editor.
+		useEffect( () => {
+			setRenderingMode( settings.defaultRenderingMode ?? 'post-only' );
+		}, [ settings.defaultRenderingMode, setRenderingMode ] );
```

Comment:
Why does setRenderingMode exist? Why not reuse settings?

Reply:
because it can be changed within the editor itself. the "settings" are something an external user of the `EditorProvider` component provides.